### PR TITLE
[sec-approval/2] add a basic acceptance test suite (bug 1550013)

### PR DESCRIPTION
**Pull request by mars-f**
Add a basic acceptance test suite that mocks Lando API responses and can
 validate Lando UI's HTML output.

### CODE REVIEW

**Comment by purelogiq** [comment]
Last I was working on Lando UI we decided to not create these types of tests because they can get overly complicated and since lando-api is constantly changing they can get stale. Unit tests + visual inspection was deemed enough.
That was last year though, so maybe things have changed. I'll defer to @smacleod on this one.

File: tests/conftest.py [hunk 1]
```diff
@@ -25,11 +25,15 @@ def docker_env_vars(monkeypatch):
...
 
+    # FIXME: this logic branch needs a comment to explain why we don't use the
+    #  local env.
     if in_circleci():
         import socket
         monkeypatch.setenv('DEBUG', 'True')
         monkeypatch.setenv('HOST', '0.0.0.0')
         monkeypatch.setenv('PORT', '7777')
+        # FIXME: I need this variable set for pytest+pycharm. Why not set it
+        #  for all envs?
```
Comment (smacleod) [inline tests/conftest.py]: I believe this was all due to the port from circle CI 1.0 to 2.0. The way Dylan handled that meant that `docker-compose` wasn't being used to run the tests (see https://github.com/mozilla-conduit/lando-ui/blob/master/.circleci/config.yml).

I made things a little simpler in Lando API by unifying the prod and dev containers, but didn't get around to this for lando UI. For more context see https://github.com/mozilla-conduit/lando-api/commit/9e94099e5adac8fd2760246beca686ff659256a5#diff-1d37e48f9ceff6d8030570cd36286a61 and https://phabricator.services.mozilla.com/D6085

File: tests/test_sec_approval_workflow.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import time
+from copy import deepcopy
+from unittest.mock import patch
+
+import pytest
+
+LANDO_API_REVISION = {
```
Comment (smacleod) [inline tests/test_sec_approval_workflow.py]: Why `...REVISION` when this is a stack?

File: tests/test_sec_approval_workflow.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+
+def test_sec_approval_workflow_mark_shown_if_revision_is_secure(
+    client, authenticated_session, apidouble
+):
+    apidouble.side_effect.stack_response["revisions"][0]["is_secure"] = True
+    rv = client.get("/D1/")
+    assert rv.status_code == 200
+    assert (
+        b"<!-- This revision is subject to the sec-approval workflow -->" in
```
Comment (smacleod) [inline tests/test_sec_approval_workflow.py]: This type of testing where we're checking the html output could get pretty fragile IMO. I worry the cost of repeatedly updating these greatly outweighs the value they bring.

**Pull request merged**